ICSharpCode.Decompiler/IL/ControlFlow/ControlFlowSimplification.cs

```diff
@@ -101,6 +102,40 @@ bool IsSimple(ILInstruction inst)
 			}
 		}
 
+		private static bool IsEmpty(ILInstruction inst) =>
+			inst is Nop || inst is Block block && block.Instructions.Count == 0 && block.FinalInstruction is Nop;
+
+		// Replaces dead code with NOPs
+		// if (...) nop; [else nop;] -> nop;
+		// if (...) { x; } x; -> x;
+		// if (...) { x; } else { x; } -> x;
+		// TODO: FIXME: Extra processing required to eliminate condition dead code and dead stack stores
+		private static void RemoveDeadIfs(ILFunction function, ILTransformContext context)
+		{
+			foreach (var block in function.Descendants.OfType<Block>()) {
+				for (int i = block.Instructions.Count - 1; i >= 0; i--) {
+					if (block.Instructions[i] is IfInstruction ifInst) {
+						if (IsEmpty(ifInst.TrueInst) && IsEmpty(ifInst.FalseInst)) {
+							block.Instructions[i].ReplaceWith(new Nop().WithILRange(ifInst));
+						} else {
+							var t = ifInst.TrueInst;
+							var f = ifInst.FalseInst;
+							var n = i < block.Instructions.Count - 1 ? block.Instructions[i + 1] : null;
+							if (t.ToString().CompareTo(f.ToString()) == 0) {
```

Comment:
`ILInstruction.ToString()` is only meant for debug output and not for comparisons.